mobility_games/games/semantic_waypoints.py

```diff
@@ -0,0 +1,292 @@
+            if tag_id in self.tag_messages:
+                self.tag_messages[tag_id].append(message)
+            else:
+                messages = []
+                messages.append(message)
+                self.tag_messages[tag_id] = messages
+
+    # Displays all messages at AR Tag with tag_id
+    def display_messages(self, tag_id):
+        if tag_id in self.tag_messages:
+            print "Someone has left you a message:"
+            self.engine.say('Somone has left you a message')
+            for message in self.tag_messages[tag_id]:
+                print message + '\n'
+                self.engine.say(message)
+
+    def update_destination_pose(self):
+        if self.visited_tags_calibrate[0] == self.tag_id:
+            ARFrame = 'AR'
+        else:
+            ARFrame = "AR_" + str(self.tag_id);
+
+        if (self.listener.frameExists("odom")
+                and self.listener.frameExists(ARFrame)):
+
+            try:
+                t = self.listener.getLatestCommonTime("odom", ARFrame)
+
+                # get transform to make tag_frame (parent) & odom (child)
+                trans, rot = self.listener.lookupTransform(
+                        "odom", ARFrame, t)
+
+                self.translations = trans
+                # self.rotations = rot
+
+            except (tf.ExtrapolationException,
+                    tf.LookupException,
+                    tf.ConnectivityException) as e:
+                print e
+
+
+    # Starts new game 
+    #   - Prompt user to enter destination
+    #   - Restart timer
+    #   - End game by typing 'done'
+    def start_new_game(self):
+        self.engine.say("What would you like to find?")
+        search_tag_name = raw_input("What would you like to find? ")
+        if search_tag_name in self.tag_name_to_id:
+            confirm_msg = "Looking for " + search_tag_name
+            self.engine.say(confirm_msg)
+            self.tag_id = self.tag_name_to_id[search_tag_name]
+            
+            self.update_destination_pose()
+           
+            self.start_run = True
+            self.start_time = time.time()
+        elif search_tag_name == 'done':
+            print "Goodbye!"
+        else:
+            err_msg =  '"' + search_tag_name + '" does not exisit. Try again!'
+            self.engine.say(err_msg)
+            print err_msg
+            self.start_new_game()
+
+    def finish_run(self, tag_id):
+        system('aplay ' + path.join(self.sound_folder, 'ding.wav'))
+        self.start_run = False
+        confirm_msg = "Found " + self.tag_id_to_name[tag_id]
+        print confirm_msg
+        self.engine.say(confirm_msg)
+        # print "Distance to Destination: " + str(self.distance_to_destination)
+
+        self.visited_tags_run.append(tag_id) 
+        self.end_time = time.time()    
+        elapsed_time = self.end_time - self.start_time
+        time_msg = "Took " + str(round(elapsed_time, 2)) + " seconds!"
```

Comment:
This is a little nicer to look at, and doesn't require the `str` operator:

`time_msg = "Took %s seconds!" % round(elapsed_time, 2)`